website/templates/search_bar.mako

```diff
@@ -14,3 +15,31 @@
         </div>
     </div>
 </div>
+
+<script>
+/** Only for Search Bar Placeholder -- Allow IE and other browsers to work as the same */
+function placeholder(inputDom, inputLabel) {
+    inputDom.on('input', function () {
+        if (inputDom.val() === '') {
+            inputLabel.css( "visibility", "visible" );
+        } else {
+            inputLabel.css( "visibility", "hidden" );
+        }
+    });
+}
+$(document).ready(function() {
+    var inputDom =  $("#searchPageFullBar");
+    var inputLabel =  $('#searchBarLabel');
+    placeholder(inputDom, inputLabel);
+    inputDom.focus();
+
+    //Make sure IE cursor is located at the end of text
+    var $inputVal = inputDom.val();
+    inputDom.val('').val($inputVal);
+
+    //For search page with existing input, make sure placeholder is hidden.
+    if(inputDom.val() !== '' ){
+         inputLabel.css( "visibility", "hidden" );
```

Comment:
It looks like all this logic can go into placeholder function (which should be renamed to placeholderShim()).